src/protocol/sort/reveal.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    error::BoxError,
+    field::Field,
+    helpers::{
+        mesh::{Gateway, Mesh},
+        prss::SpaceIndex,
+        Direction,
+    },
+    protocol::{context::ProtocolContext, RecordId, Step},
+    secret_sharing::Replicated,
+};
+use embed_doc_image::embed_doc_image;
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've revealed their own shares
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct AValue<F> {
+    a: F,
+}
+
+/// This implements reveal algorithm of "Efficient Secure Three-Party Sorting Protocol with an Honest Majority" at communication cost of 3R.
+/// Revealing is the protocol that on input a pair of shares, outputs a secret. For simplicity, we
+/// consider a simple revealing in which each Pi sends [a]i to Pi+1 and then reconstructs a from [a]i and
+/// [a]i−1.
+// Input: Each helpers know their own secret shares
+// Output: At the end of the protocol, all 3 helpers know a revealed (or opened) secret
+#[derive(Debug)]
+pub struct Reveal<F> {
+    input: Replicated<F>,
+}
+
+impl<F: Field> Reveal<F> {
+    #[allow(dead_code)]
+    pub fn new(input: Replicated<F>) -> Self {
+        Self { input }
+    }
+
+    /// Steps
+    /// Each helper sends their left share to the right helper. The helper then reconstructs their secret by adding the three shares
+    /// i.e. their own shares and received share.
+    #[embed_doc_image("reveal", "images/sort/reveal.png")]
```

Comment:
I think `H3` initial state is wrong - it should know (`a3`, `a1`) instead of (`a2`, `a3`)

Reply:
My bad. Fixed now